core/services/relay/evm/chain_writer.go

```diff
@@ -155,8 +162,44 @@ func (w *chainWriter) GetTransactionStatus(ctx context.Context, transactionID uu
 	return commontypes.Unknown, fmt.Errorf("not implemented")
 }
 
+// SubmitTransaction
+//
+// Returns the execution and data availability (L1Oracle) fees for the chain.
+// Dynamic fees (introduced in EIP-1559) include a fee cap and a tip cap. If the dyanmic fee is not available,
+// (if the chain doesn't support dynamic TXs) the legacy GasPrice is used.
 func (w *chainWriter) GetFeeComponents(ctx context.Context) (*commontypes.ChainFeeComponents, error) {
-	return nil, fmt.Errorf("not implemented")
+	if w.ge == nil {
+		return nil, fmt.Errorf("not implemented")
+	}
+
+	gasPriceWei, _, err := w.ge.GetFee(ctx, nil, 0, assets.NewWei(w.maxGasPrice))
```

Comment:
I understand this was copy pasted, but I find the usage of `gasPriceWei` quite confusing since what is returned is not a gas price value but rather an `EvmFee` object instead.

Reply:
Good point, I renamed this object `evmFee`